client/src/api/UserAPI.ts

```diff
@@ -1,54 +1,131 @@
 import axios, { AxiosResponse } from 'axios';
 
 import {
+  UserInfo,
   SignupVals,
   SignupResponse,
   LoginVals,
   LoginResponse,
+  ResetPasswordVals,
+  MessageResponse,
+  UpdateUsernameVals,
+  UpdatePasswordVals,
 } from '../interfaces/interfaces';
 
 /**
  * API signup request
  *
- * @param {string} username - Username
- * @param {string} password - Password
- * @param {string} securityQuestion - Security Question
- * @param {string} securityAnswer - Security Answer
+ * @param {dict} data - Username
+ *                    - Password
+ *                    - Security Question
+ *                    - Security Answer
  * @returns {AxiosResponse}
- * @throws {AxiosError}
  */
 export const signupRequest = async (
   data: SignupVals,
 ): Promise<AxiosResponse<SignupResponse>> => {
-  try {
-    const response: AxiosResponse = await axios.post<SignupResponse>(
-      'http://localhost:7005/users/register',
-      data,
-    );
-    return response;
-  } catch (error) {
-    throw error;
-  }
+  const response: AxiosResponse = await axios.post<SignupResponse>(
+    'http://localhost:7005/users/register',
+    data,
+  );
+  return response;
 };
 
 /**
  * API login request
  *
- * @param {string} username - Username
- * @param {string} password - Password
+ * @param {dict} data - Username
+ *                    - Password
  * @returns {AxiosResponse}
- * @throws {AxiosError}
  */
 export const loginRequest = async (
   data: LoginVals,
 ): Promise<AxiosResponse<LoginResponse>> => {
-  try {
-    const response = await axios.post<LoginResponse>(
-      'http://localhost:7005/users/login',
-      data,
-    );
-    return response;
-  } catch (error) {
-    throw error;
-  }
+  const response = await axios.post<LoginResponse>(
+    'http://localhost:7005/users/login',
+    data,
+  );
+  return response;
+};
+
+/**
+ * API Reset Password
+ *
+ * @param {dict} data - Username
```

Comment:
the parameter type here does not match the one actually being accepted. I think it makes more sense with 
`@param {ResetPasswordVals} data - values for reset password`. should be fixed for the rest of the API calls as it is not type dict